examples/energy-monitor/ui/mid_main.slint

```diff
@@ -86,7 +87,45 @@ export component MidMain {
         }
     }
 
+    i-overlay := Rectangle {
+        in-out property <int> index;
+
+        visible: false;
+        opacity: 0.0;
+        width: 100%;
+        height: 100%;
+        background: Theme.palette.dimmer;
+
+        i-overlay-container := PageContainer {
+            x: (root.width - self.width) / 2;
+            y: (root.height - self.height) / 2;
+            width: parent.width - 20px;
+            height: parent.height - 20px;
```

Comment:
@tronical 

> I wonder if it's somehow possible to implement this by means of "zooming in", i.e. giving the clicked on item the size of the screen?

It's not trivial, because it's inside of a layout, the layout is inside of the flickable, ... . The problem is at the moment tree operations are limited in Slint. What would helps a lot in that context would be to change the parent of the element e.g. to an overlay and on close to move it back to the origin parent. 

An other approach I checked was to create the overlay from the position and size of the clicked item, but it end up by a lot of ugly code to get the correct x, y. A possibility to read the absolute x, y position could help on this side.

Reply:
What you can do is to have the inner screen grow outside of its parent.

Straw man example:  https://slint-ui.com/snapshots/master/editor/?snippet=import+%7B+Button%2C+VerticalBox+%2C+ScrollView%2C+HorizontalBox%7D+from+%22std-widgets.slint%22%3B%0A%0A%0Acomponent+Compo+%7B%0A++++width%3A+100px%3B%0A++++height%3A+100px%3B%0A++++out+property+%3Cbool%3E+big+%3A+false%3B%0A++++Rectangle+%7B%0A++++++++background%3A+red%3B%0A++++++++border-color%3A+blue%3B%0A++++++++border-width%3A+2px%3B%0A++++++++width%3A+big+%3F+200px+%3A+parent.width%3B%0A++++++++height%3A+big+%3F+200px+%3A+parent.height%3B%0A++++++++animate+width%2C+height+%7B+duration%3A+100ms%3B+%7D%0A++++%7D%0A++++TouchArea+%7B+clicked+%3D%3E+%7Bbig+%3D+%21big%3B+%7D+%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++width%3A+320px%3B%0A++++height%3A+240px%3B%0A++++property+%3Cbool%3E+active%3A+c1.big+%7C%7C+c2.big+%7C%7C+c3.big+%7C%7C+c4.big+%7C%7C+c5.big+%7C%7C+c6.big%3B%0A++++ScrollView+%7B+%0A++++++++HorizontalBox+%7B+%0A++++++++++++c1+%3A%3D+Compo%7B+opacity%3A+active+%26%26+%21self.big+%3F+0+%3A+1%3B+animate+opacity+%7B+duration%3A+100ms%3B+%7D+%7D%0A++++++++++++c2+%3A%3D+Compo%7B+opacity%3A+active+%26%26+%21self.big+%3F+0+%3A+1%3B+animate+opacity+%7B+duration%3A+100ms%3B+%7D+%7D%0A++++++++++++c3+%3A%3D+Compo%7B+opacity%3A+active+%26%26+%21self.big+%3F+0+%3A+1%3B+animate+opacity+%7B+duration%3A+100ms%3B+%7D+%7D%0A++++++++++++c4+%3A%3D+Compo%7B+opacity%3A+active+%26%26+%21self.big+%3F+0+%3A+1%3B+animate+opacity+%7B+duration%3A+100ms%3B+%7D+%7D%0A++++++++++++c5+%3A%3D+Compo%7B+opacity%3A+active+%26%26+%21self.big+%3F+0+%3A+1%3B+animate+opacity+%7B+duration%3A+100ms%3B+%7D+%7D%0A++++++++++++c6+%3A%3D+Compo%7B+opacity%3A+active+%26%26+%21self.big+%3F+0+%3A+1%3B+animate+opacity+%7B+duration%3A+100ms%3B+%7D+%7D%0A++++++++++++%0A++++++++%7D%0A++++%7D%0A%7D%0A

Reply:
I had a different approach in mind but it would be difficult to implement:

https://firebasestorage.googleapis.com/v0/b/design-spec/o/projects%2Fm3%2Fimages%2Fl78agsqx-02%20-%20Container%20transform%20-%203P.mp4?alt=media&token=e11c41fb-8a53-4af9-8e62-497705fa372d

But @ogoffart I think you suggestion would fit well. I will try it out thank you.